pkg/pool-stable/test/ComposableStablePool.test.ts

```diff
@@ -1537,6 +1538,298 @@ describe('ComposableStablePool', () => {
           expect(virtualSupply).to.be.equalWithError(senderBptBalance, 0.0001);
         });
       });
+
+      context('nested pools (beethovenxfi: PR 1671)', () => {
```

Comment:
I'd consider placing this in a separate file, along with other complex scenarios we might produce, to try to keep things tidy. The sheer size of this file makes it hard to navigate.

Reply:
@TomAFrench do you want me to move this out (since I put it in?) I actually started it as a separate file, but then didn't want to duplicate all the initial setup. We could put the others in the new file too (the rate-lowering one would still fail).